File: src/main/java/com/telegram/codex/integration/telegram/application/port/in/TelegramMessageParser.java
package com.telegram.codex.integration.telegram.application.port.in;

import com.telegram.codex.integration.telegram.domain.InboundMessage;
import com.telegram.codex.integration.telegram.domain.webhook.TelegramUpdate;

public interface TelegramMessageParser {

    InboundMessage parseIncomingTelegramMessage(TelegramUpdate update);
}

File: src/main/kotlin/com/telegram/codex/integration/telegram/application/port/in/TelegramMessageParser.kt
package com.telegram.codex.integration.telegram.application.port.`in`

import com.telegram.codex.integration.telegram.domain.InboundMessage
import com.telegram.codex.integration.telegram.domain.webhook.TelegramUpdate

interface TelegramMessageParser {
    fun parseIncomingTelegramMessage(update: TelegramUpdate?): InboundMessage?
}
